docs/deploy-configure.md

```diff
@@ -132,3 +134,76 @@ an example of how to provide the required configuration for a MongoDB
 replica set.
 
 See [here](../deploy/openshift/operator_openshift.yaml) for an example of how to configure the Operator deployment.
+
+## Define a Custom Database Role
+
+You can define [custom roles](https://docs.mongodb.com/manual/core/security-user-defined-roles/) to give you fine-grained access control over your MongoDB database resource.
+
+  **NOTE**: Custom roles are scoped to a single MongoDB database resource.
+
+To define a custom role:
+
+1. Add the following fields to the MongoDB resource definition:
+
+   | Key | Type | Description | Required? |
+   |----|----|----|----|
+   | `spec.security.roles` | array | Array that defines [custom roles](https://docs.mongodb.com/manual/core/security-user-defined-roles/) roles that give you fine-grained access control over your MongoDB deployment. | Yes |
+   | `spec.security.roles.role` | string | Name of the custom role. | Yes |
+   | `spec.security.roles.db` | string | Database in which you want to store the user-defined role. | Yes |
+   | `spec.security.roles.authenticationRestrictions` | array | Array that defines the IP address from which and to which users assigned this role can connect. | No |
+   | `spec.security.roles.authenticationRestrictions.clientSource` | array | Array of IP addresses or CIDR blocks from which users assigned this role can connect. <br><br> MongoDB servers reject connection requests from users with this role if the requests come from a client that is not present in this array. | No |
+   | `spec.security.roles.authenticationRestrictions.serverAddress` | array | Array of IP addresses or CIDR blocks to which users assigned this role can connect. <br><br> MongoDB servers reject connection requests from users with this role if the client requests to connect to a server that is not present in this array. | No |
+   | `spec.security.roles.privileges` | array | List of actions that users granted this role can perform. For a list of accepted values, see [Privilege Actions](https://docs.mongodb.com/manual/reference/privilege-actions/#database-management-actions) in the MongoDB Manual for the MongoDB versions you deploy with the Kubernetes Operator. | Yes |
+   | `spec.security.roles.privileges.actions` | array | Name of the role. Valid values are [built-in roles](https://docs.mongodb.com/manual/reference/built-in-roles/#built-in-roles). | Yes |
+   | `spec.security.roles.privileges.resource` | collection | Resources for which the privilege `spec.security.roles.privileges.actions` apply. <br><br> This collection must include either: <br><br> <ul><li>The `spec.security.roles.privileges.resource.database` and `spec.security.roles.privileges.resource.collection` settings, or</li><li>The `spec.security.roles.privileges.resource.cluster` setting with a value of `true`.</li></ul> | Yes |
```

Comment:
Is this entry really just the prefix for the next three? I'm not clear how it would be used on it's own.

Reply:
it is. it wouldn't be used on it's own. you must include either .database and .collection or .cluster. Do you have suggestions on how we can clarify the text here?

Reply:
It's a little tricky because the 'yes' in the required column make it look like it is its own entity. 

One option is to leave it out and note in the other three that an appropriate combination is required. 

If we keep this entry, maybe change the 'Yes' to an asterix or 'n.a.' and modify the text slightly so instead of 'This collection must include either' it reads 'This field is set using either the cluster or the database and collection settings listed below.'

Reply:
I think the idea to remove this entry is a good one. I have already described the appropriate combinations in the descriptions of its children.